### PR TITLE
feat: extensiblity types

### PR DESCRIPTION
**Pull request by jelz**
## Summary

Sync types with API schemas

## Checklist (check all before merging)

- [ ] Both unit and integration tests are passing
- [ ] There are no breaking changes
- [ ] Changes are reflected in the documentation

When adding a new method:

- [ ] The new method is exported through the default and plain CMA client
- [ ] All new public types are exported from `./lib/export-types.ts`
- [ ] Added a unit test for the new method
- [ ] Added an integration test for the new method
- [ ] The new method is added to the documentation

### CODE REVIEW

File: lib/entities/app-definition.ts [hunk 1]
```diff
@@ -72,11 +43,11 @@ export type AppDefinitionProps = {
   /**
    * URL where the root HTML document of the app can be found
    */
-  src: string
+  src?: string
   /**
    * Locations where the app can be installed
    */
-  locations: AppLocation[]
+  locations?: AppLocation[]
```
Comment (Jwhiles) [inline lib/entities/app-definition.ts:50]: It looks like locations is always defined, but can be empty.

Comment (Jwhiles) [reply lib/entities/app-definition.ts:50]: Or at least that's the case when you update through the user interface

File: lib/entities/editor-interface.ts [hunk 2]
```diff
@@ -8,24 +8,58 @@ import * as endpoints from '../plain/endpoints'
...
-  widgetId: string
-  widgetNamespace: string
   settings?: Record<string, any>
 }
 
 export interface Editor {
+  /**
+   * Type of the widget used
+   */
+  widgetNamespace: string
```
Comment (Jwhiles) [inline lib/entities/editor-interface.ts:32]: Should we limit this to the specific values that are possible?

**Review by suevalov — APPROVED** [review]
Splendid!

P.S. Should it be `fix: extensibility types` or `feat: extensibility types`?

**Review by Jwhiles — APPROVED** [review]

**Comment by jelz** [comment]
@suevalov done, but I still cannot merge, can you?

**Pull request merged**